### DISCUSSION TITLE
Azure AD app permissions for getting all sites using PnP Commands

### DISCUSSION THREAD

**Discussion by Abhishekbaskar26** [pnp/powershell#3472] [Q&A]
Hi
I am connecting to SharePoint admin center using command "Connect-PnPOnline" with certificate thumbprint and Client Id. 
In my Azure AD App, I am having the SharePoint Permission "Sites.Read.All". I am trying to get all the site collections from tenant using this PnP command "Get-PnPTenantSite". I am not able to get any data. It is showing me an Unauthorized Operation.
![image](https://github.com/pnp/pnpcore/assets/127750001/3241b947-c2b5-406c-87be-e33bc85f07fb)

If i am using "Sites.Manage.All" in Azure AD App, I can able to do all operations.
Can you please explain me Why sites.Read.All is not enough for getting the site collection Details?

**Comment by jansenbe** [comment]
Think you also need to grant the equivalent Graph permission

**Reply by Abhishekbaskar26** [reply]
I gave the Graph permission for Site as well. Not Working with Read.All in both SharePoint and Graph.

**Reply by jansenbe** [reply]
@Abhishekbaskar26 : I checked the code for that cmdlet and it's not using PnP Core SDK but just plain CSOM. I'll transfer this discussion to the PnP PowerShell repo as this problem is not related to PnP Core SDK.
